addon/components/bs-modal.js

```diff
@@ -679,10 +686,12 @@ export default class Modal extends Component {
       return;
     }
 
-    if (this.args.open !== false) {
-      this.show();
-    } else {
-      this.hide();
-    }
+    next(() => {
```

Comment:
Why is the next scheduling necessary? Might this be responsible for the failure in FastBoot?

Reply:
This prevents `state` being modified after it has been used already. Without scheduling the work for the next runloop we run into this deprecation:

> DEPRECATION: You attempted to update `state` on `wrapperClass`, but it had already been used previously in the same computation.  Attempting to update a value after using it in a computation can cause logical errors, infinite revalidation bugs, and performance issues, and is not supported.

I think this is a common issue if a method called by a `ember-render-helpers` or `ember-render-modifiers` sets a property. And the reason why ember-render-helpers and ember-render-modifiers should be considered a footgun.

I'm not 100%ly sure why this is happening. But I _think_ the helper / modifier is not execute as part of the `actions` queue. So it must delay all changes to state for the next runloop.

Fastboot test is still failing without scheduling it for the next runloop. I _guess_ some FastBoot guards prevent the modal from being rendered. But I haven't fully looked into. Debugging FastBoot issues is way more painful than other stuff. :disappointed:

Reply:
You were right.

1. `getOwner(this).isDestroyed` was `true` in `next()` callback. Seems as if application gets destroyed before.
2. The next runloop had only hidden the assertion but not resolved the issue. Tracked property `state` was read in `show` and `hide` method before it was set.

Refactored to have `state` property not tracked as it must be read by `show` and `hide` modal. Otherwise we can not exit `show` method if the modal is already shown. Trying to show a modal with all animations, which is already shown, would lead to strange bugs.

Introduced another tracked property `isModalShown`, which is managed by `state` through a setter.

Actually this duplicated the state. But it seems that this is required by tracked properties if you must read the state before setting it.